### PR TITLE
feat: add deploying→deployed/blocked reconciliation via GH Actions run outcome

### PR DESCRIPTION
**Pull request by dmcaulay**
## Summary
- Adds a `reconcileDeployingTasks` pass to `pr-state-reconciler.ts` that reconciles tasks stuck at `status: "deploying"` (e.g. because the `/shipwright:deploy` session that would normally transition them died, timed out, or was evicted mid-poll)
- Resolves the linked PR's merge commit SHA, fetches GH Actions runs at that SHA, and checks for a `Deploy`-named run's actual `conclusion` — never uses "PR merged" alone as the success signal
- Falls back to classifying post-merge CI run conclusions on the same SHA when no `Deploy`-named run exists (mirrors `deploy.md` Step 5c), so repos without a Deploy workflow don't get stuck permanently
- Marks the task `blocked` with a note citing the failed run ID on deploy/CI failure; leaves in-progress or zero-run cases untouched (not yet resolvable)

## Acceptance Criteria
| # | Criterion | Status |
|---|-----------|--------|
| 1 | New pass reconciles deploying tasks via actual GH Actions run outcome for merge commit — never uses "PR merged" alone | MET |
| 2 | No-Deploy-workflow fallback checks post-merge CI on same SHA before deciding | MET |
| 3 | Failed deploy runs mark task blocked with note citing the run | MET |
| 4 | Test file extended with fake GH Actions responses covering success/failure/fallback-success/in-progress | MET |

## Test Plan
- `bun test src/pr-state-reconciler.unit.test.ts` — 51 tests pass (12 new for the deploying-task pass, covering: successful Deploy run → deployed, failed Deploy run → blocked, fallback CI success → deployed, fallback CI failure → blocked, in-progress run → untouched, zero runs → untouched, failed-then-succeeded retry → deployed, plus defensive skip/error-isolation cases)
- `bun run typecheck` — clean across all workspaces
- `bunx biome lint .` — clean
- [x] Pre-commit checks passing

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: agent/src/pr-state-reconciler.ts [hunk 1]
```diff
@@ -323,15 +354,144 @@ export async function reconcilePrOpenTasks(deps: PrStateReconcilerDeps): Promise
...
+ * be masked by an earlier failed attempt at the same SHA); with no success
+ * present, any still-incomplete run means the outcome isn't resolvable yet;
+ * only once every run has completed with none succeeding is the outcome a
+ * genuine failure. Zero runs is "pending" — nothing to decide from yet.
+ */
+function classifyRuns(
+  runs: Array<{ status: string; conclusion: string | null }>,
+): RunOutcome {
+  if (runs.length === 0) return "pending";
+  if (runs.some((r) => r.conclusion === "success")) return "success";
```
Comment (dodizzle) [inline agent/src/pr-state-reconciler.ts:375]: `classifyRuns` is shared between the primary path (runs filtered to `name === "Deploy"` — multiple entries are plausible retries of the same workflow, so "any success wins" makes sense) and the fallback path (all runs at the SHA, unfiltered by name — i.e. distinct, unrelated workflows like `CI`, `Security Scan`, `Docs Build`). Because this `.some(conclusion === "success")` check short-circuits before checking for failures, a mix of one successful and one genuinely-failed *distinct* workflow at the same SHA gets classified as `"success"` in the fallback path — marking the task `deployed` even though a real CI job failed on that commit. This diverges from `deploy.md` Step 5c (which this PR's docstring says it mirrors): Step 5c requires `conclusion == "success"` for *every* run seen, and treats a mix as failure. Worth a stricter fallback-specific classifier, or a flag distinguishing the two callers — no test currently covers the mixed-distinct-workflow scenario.

**Review by dodizzle — COMMENTED** [review]
Verdict: COMMENT — classifyRuns() reuses "any success wins" logic (correct for same-workflow Deploy retries) in the fallback path where runs are unfiltered distinct workflows, so a mixed success+failure result there is misclassified as "success" instead of matching deploy.md Step 5c's "any run fails → blocked" rule. Everything else (primary path, error isolation, test coverage) looks solid.

**Review by dmcaulay — COMMENTED** [review]
Verdict: APPROVE — The prior fallback-path classification bug is fixed correctly and completely: `classifyRunsAnySuccessWins` (Deploy-retry path) and `classifyRunsAllMustSucceed` (distinct-workflow fallback path) now match deploy.md Step 5c's semantics exactly, backed by a dedicated regression test for the mixed-workflow case. Edge cases (missing repo/PR/commitSha, in-progress, zero runs, retry-after-failure, error isolation) are all individually tested with injected fakes. No new issues found. Posting as COMMENT since the PR author matches the reviewing account (GitHub blocks self-APPROVE via the API) — the recorded verdict above is APPROVE.
